src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQueryEnvironment.java

```diff
@@ -343,14 +344,45 @@ protected CqueryNode getValueFromKey(SkyKey key) throws InterruptedException {
    */
   private ImmutableList<CqueryNode> getConfiguredTargetsForLabel(Label label)
       throws InterruptedException {
-    ImmutableList.Builder<CqueryNode> ans = ImmutableList.builder();
-    for (BuildConfigurationValue config : transitiveConfigurations.values()) {
-      CqueryNode kct = getConfiguredTarget(label, config);
-      if (kct != null) {
-        ans.add(kct);
+    var ans = ImmutableList.<CqueryNode>builder();
```

Comment:
It looks like we're largely cloning the logic from [`getConfiguredTarget`](https://github.com/bazelbuild/bazel/blob/ec1aea0e06c8bb63af323799dbe249bbe368f6fc/src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQueryEnvironment.java#L302-L317), which this method originally delegated to?

What's the advantage to adding the logic here vs. there? `getConfiguredTaget` has a few other callers that could get this adjusted behavior. Would that be good or bad?

Reply:
That would be bad: `getConfiguredTarget` should return a `ConfiguredTarget` in a given configuration, *not* a delegated one, if available. The current method should return each `ConfiguredTarget` for a given `Label` exactly once. I don't think any of these methods can reuse (parts of) the other.